src/core/vector/qgsvectorlayerprofilegenerator.cpp

```diff
@@ -750,6 +750,18 @@ bool QgsVectorLayerProfileGenerator::generateProfile( const QgsProfileGeneration
   mProfileCurveEngine.reset( new QgsGeos( mProfileCurve.get() ) );
   mProfileCurveEngine->prepareGeometry();
 
+  if ( mTolerance == 0.0 ) // geos does not handle very well buffer with 0 size
+  {
+    mProfileBox = std::unique_ptr<QgsAbstractGeometry>( mProfileCurve->clone() );
```

Comment:
still valid

Reply:
By reviewing my own code I just see I, now, use `mProfileBufferedCurveEngine` nearly everywhere instead of having `if ( mTolerance == 0.0 )` and have to handle both cases each time: I do it once in `generateProfile`. 

I still use `mProfileCurveEngine` once when the polygon intersects a line and I use the tesselation. But in all other case I use the `mProfileBufferedCurveEngine`.  May be the name is not the best...